thavalon-webapp/src/components/gameComponents/interactions/missionManager.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from "react";
+import ReactModal from "react-modal";
+import { GameSocket, InboundMessage, InboundMessageType } from "../../../utils/GameSocket";
+import { GameActionType, InteractionProps, MissionCard, Vote, MissionGoingMessage, GameMessage, GameMessageType, MissionResultsMessage } from "../constants";
+import { createSelectedPlayerTypesList, sendGameAction } from "../gameUtils";
+import { PlayerCard } from "../playerCard";
+
+/**
+ * The required properties for the MissionManager
+ */
+interface MissionManagerProps extends InteractionProps {
+    me: string,
+    message: MissionGoingMessage,
+    votes: Map<string, Vote>,
+}
+
+/**
+ * The required properties for the MissionCard
+ */
+interface MissionCardProps {
+    submitMissionCard(card: MissionCard): void
+}
+
+/**
+ * Required properties for the MissionResult modal.
+ */
+interface MissionResultModalProps {
+    message: MissionResultsMessage | undefined,
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+/**
+ * Component managing all of the mission related interactions.
+ * @param props Required properties for the MissionManager
+ */
+export function MissionManager(props: MissionManagerProps): JSX.Element {
+    useEffect(() => {
+        const connection = GameSocket.getInstance();
+        connection.onGameEvent.subscribe(handleMessage);
+        return () => connection.onGameEvent.unsubscribe(handleMessage);
+    });
+
+    /**
+     * Handles any incoming server message
+     * @param message A message from the server
+     */
+    function handleMessage(message: InboundMessage): void {
+        // TODO: Support cardPlayed message types here.
+        if (message.messageType !== InboundMessageType.GameMessage) {
+            return;
+        }
+    }
+
+    // State maintaining if the player has played a card or not
+    const [hasPlayedCard, setHasPlayedCard] = useState(false);
+
+    /**
+     * Submits a mission card or QB to the server and updates if the player
+     * has played a card accordingly.
+     * @param card A mission card to play
+     */
+    function submitMissionCard(card: MissionCard): void {
+        if (card === MissionCard.QuestingBeast) {
+            sendGameAction(GameActionType.QuestingBeast);
+        } else {
+            sendGameAction(GameActionType.Play, { card: card });
```

Comment:
more of a backend comment, but technically playing a questing beast is playing a card. I feel this should be one backend action instead of 2 for when we want to support games with questing beast and pelinor

Reply:
I agree, but this would need a @bnavetta to review. For now, I'm keeping it since it's such a small thing.

Reply:
I separated them originally because on the backend it's a little annoying to use the same type for QuestingBeast cards along with regular mission cards, but we could probably combine them.